test-e2e/sync.js

```diff
@@ -925,51 +940,100 @@ test('data sync state is properly updated as data sync is enabled and disabled',
     invitorProject.$sync.getState().data.isSyncEnabled,
     'after enabled sync, data sync is enabled for local device'
   )
+  assert.deepEqual(
+    invitorProject.$sync.getState().remoteDeviceSyncState,
+    {
+      [invitees[0].deviceId]: {
+        initial: { isSyncEnabled: true, want: 0, wanted: 0 },
+        data: { isSyncEnabled: false, want: 1, wanted: 0 },
+      },
+      [invitees[1].deviceId]: {
+        initial: { isSyncEnabled: true, want: 0, wanted: 0 },
+        data: { isSyncEnabled: false, want: 1, wanted: 0 },
+      },
+    },
+    'data sync is still disabled for remote peers'
+  )
 
-  Object.values(invitorProject.$sync.getState().remoteDeviceSyncState).forEach(
-    (syncState) => {
-      assert.ok(
-        !syncState.data.isSyncEnabled,
-        'data sync is still disabled for all remote devices'
-      )
+  const inviteeAppearsEnabledPromise = pEvent(
+    invitorProject.$sync,
+    'sync-state',
+    ({ remoteDeviceSyncState }) =>
+      remoteDeviceSyncState[invitees[0].deviceId]?.data.isSyncEnabled
+  )
+  const invitorProjectSyncedWithFirstInviteePromise = pEvent(
+    invitorProject.$sync,
+    'sync-state',
+    ({ remoteDeviceSyncState }) => {
+      const remoteData = remoteDeviceSyncState[invitees[0].deviceId]?.data ?? {}
+      return remoteData.want + remoteData.wanted === 0
+    }
+  )
+  const firstInviteeProjectSyncedWithInvitorPromise = pEvent(
+    inviteesProjects[0].$sync,
+    'sync-state',
+    ({ remoteDeviceSyncState }) => {
+      const remoteData = remoteDeviceSyncState[invitor.deviceId]?.data ?? {}
+      return remoteData.want + remoteData.wanted === 0
     }
   )
 
   inviteesProjects[0].$sync.start()
 
   assert.ok(
     inviteesProjects[0].$sync.getState().data.isSyncEnabled,
-    'remote peer has data sync enabled after starting sync'
+    'invitee has data sync enabled after starting sync'
   )
 
-  await delay(2000)
-  assert.ok(
-    Object.values(invitorProject.$sync.getState().remoteDeviceSyncState).some(
-      (remoteState) => {
-        return remoteState.data.isSyncEnabled
-      }
-    ),
-    'at least one remote peer has enabled data sync'
+  await inviteeAppearsEnabledPromise
+
+  assert(
+    invitorProject.$sync.getState().remoteDeviceSyncState[invitees[0].deviceId]
+      ?.data.isSyncEnabled,
+    'one invitee has enabled data sync'
+  )
+  assert(
+    !invitorProject.$sync.getState().remoteDeviceSyncState[invitees[1].deviceId]
+      ?.data.isSyncEnabled,
+    'other invitee has not enabled data sync'
+  )
+
+  await invitorProjectSyncedWithFirstInviteePromise
+
+  await Promise.all([
+    invitorProjectSyncedWithFirstInviteePromise,
+    firstInviteeProjectSyncedWithInvitorPromise,
+  ])
```

Comment:
These promises never resolve, seemingly because stuff gets moved into `wanted` in a way that I don't think is correct...